SpatialGDK/Source/SpatialGDKServices/Private/SSpatialOutputLog.h

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "Developer/OutputLog/Private/SOutputLog.h"
+#include "HAL/FileManagerGeneric.h"
+#include "IDirectoryWatcher.h"
+#include "SlateFwd.h"
+
+DECLARE_LOG_CATEGORY_EXTERN(LogSpatialOutputLog, Log, All);
+
+// Child class of the file reader used by Unreal but with the ability to update the known file size.
+// This allows us to read a log file while it is being written to.
+class FArchiveLogFileReader : public FArchiveFileReaderGeneric
+{
+public:
+	FArchiveLogFileReader(IFileHandle* InHandle, const TCHAR* InFilename, int64 InSize, uint32 InBufferSize /*= PLATFORM_FILE_READER_BUFFER_SIZE*/)
+		: FArchiveFileReaderGeneric(InHandle, InFilename, InSize, InBufferSize)
+	{}
+	void UpdateFileSize();
+};
+
+class SSpatialOutputLog
+	: public SOutputLog
+{
+
+public:
+	SLATE_BEGIN_ARGS( SSpatialOutputLog )
+	{}
+
+	SLATE_END_ARGS()
+
+	~SSpatialOutputLog();
+
+	void Construct( const FArguments& InArgs );
+
+	TUniquePtr<FArchiveLogFileReader> CreateLogFileReader(const TCHAR* InFilename, uint32 Flags, uint32 BufferSize);
```

Comment:
Looks like this is only used internally to assign to `LogReader`. Make the assignment inside the function, remove the return value, and make this private.